mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -22,7 +22,7 @@ def log_artifact(self, local_file, artifact_path=None):
         paths = (artifact_path, file_name) if artifact_path else (file_name,)
         endpoint = posixpath.join("/", *paths)
         with open(local_file, "rb") as f:
-            resp = http_request(self._host_creds, endpoint, "PUT", data=f, timeout=600)
+            resp = http_request(self._host_creds, endpoint, "PUT", data=f)
```

Comment:
Maybe we should raise a warning message that explains how to customize the timeout length (e.g. `API request timed out. Set environment variable MLFLOW_HTTP_REQUEST_TIMEOUT (default: 120) to increase timeout.`) in `http_request` if it fails due to timeout.

Reply:
Sounds like a good idea to me :)

Reply:
Thought: 
Does a global default `MLFLOW_HTTP_REQUEST_TIMEOUT` satisfy all cases ?
e.g.  downloading model we should set timeout a bit higher. But others we don't need so long timeout.

Reply:
It's probably not ideal to use a single timeout value for all the operations, but I think it's ok.

Reply:
@dbczumar @WeichenXu123 Filed a follow-up PR: https://github.com/mlflow/mlflow/pull/6208